### PR TITLE
Fix : Klockwork fixes for unit tests

### PR DESCRIPTION
**Pull request by nbmaiti**
Signed-off-by: Nabendu Maiti <nabendu.bikash.maiti@intel.com>

### CODE REVIEW

File: tests/unit/test_RSARoutines.c [hunk 1]
```diff
@@ -415,7 +415,7 @@ TEST_CASE("rsaencrypt", "[RSARoutines][sdo]")
 	TEST_ASSERT_EQUAL_MESSAGE(-1, ret, "RSA Encryption Failed");
 
 	/* clean up */
-	if (cipher_text)
+	if (cipher_text != NULL)
```
Comment (AdithyaBaglody) [inline tests/unit/test_RSARoutines.c:418]: In a few places braces are missing.

Comment (nbmaiti) [reply tests/unit/test_RSARoutines.c]: done

File: tests/unit/test_cryptoSupport.c [hunk 2]
```diff
@@ -1878,7 +1880,7 @@ TEST_CASE("sdo_ov_verify", "[crypto_support][sdo]")
 	uint32_t signature_len = BUFF_SIZE_256_BYTES;
 	uint8_t *message_signature = malloc(signature_len);
 	TEST_ASSERT_NOT_NULL(message_signature);
-	bool *result = malloc(sizeof(bool *));
+	bool val = 1; bool *result = &val;
```
Comment (trbehera) [inline tests/unit/test_cryptoSupport.c:1883]: Shouldn't we be defining the variables in different lines?

File: tests/unit/test_cryptoSupport.c [hunk 3]
```diff
@@ -2032,7 +2033,7 @@ TEST_CASE("sdo_ov_verify_invalid_message", "[crypto_support][sdo]")
...
```
Comment (trbehera) [inline tests/unit/test_cryptoSupport.c:2036]: Same as above.

File: tests/unit/test_cryptoSupport.c [hunk 4]
```diff
@@ -2177,7 +2177,7 @@ TEST_CASE("sdo_ov_verify_invalid_message_length", "[crypto_support][sdo]")
...
```
Comment (trbehera) [inline tests/unit/test_cryptoSupport.c:2180]: Same as above.

File: tests/unit/test_cryptoSupport.c [hunk 5]
```diff
@@ -2322,7 +2321,7 @@ TEST_CASE("sdo_ov_verify_invalid_message_signature", "[crypto_support][sdo]")
...
```
Comment (trbehera) [inline tests/unit/test_cryptoSupport.c:2324]: Same as above.

File: tests/unit/test_cryptoSupport.c [hunk 6]
```diff
@@ -3453,7 +3450,7 @@ TEST_CASE("crypto_hal_sig_verify_fail_case", "[crypto_support][sdo]")
...
```
Comment (trbehera) [inline tests/unit/test_cryptoSupport.c:3453]: Same as above.

File: tests/unit/test_cryptoSupport.c [hunk 7]
```diff
@@ -2612,7 +2609,7 @@ TEST_CASE("sdo_ov_verify_invalid_pubkey", "[crypto_support][sdo]")
...
```
Comment (trbehera) [inline tests/unit/test_cryptoSupport.c:2612]: Same as above.

File: tests/unit/test_cryptoSupport.c [hunk 8]
```diff
@@ -2467,7 +2465,7 @@ TEST_CASE("sdo_ov_verify_invalid_signature_len", "[crypto_support][sdo]")
...
```
Comment (trbehera) [inline tests/unit/test_cryptoSupport.c:2468]: Same as above.

**Comment by trbehera** [comment]
We are working on fixing unit-test failure reporting. This PR can be merged once that pipeline is fixed.

File: tests/unit/test_cryptoSupport.c (hunk 2 above)
Comment (nbmaiti) [reply tests/unit/test_cryptoSupport.c]: Done

File: tests/unit/test_cryptoSupport.c (hunk 5 above)
Comment (nbmaiti) [reply tests/unit/test_cryptoSupport.c]: done

File: tests/unit/test_cryptoSupport.c (hunk 7 above)
Comment (nbmaiti) [reply tests/unit/test_cryptoSupport.c]: Done

File: tests/unit/test_cryptoSupport.c (hunk 8 above)
Comment (nbmaiti) [reply tests/unit/test_cryptoSupport.c]: done

File: tests/unit/test_cryptoSupport.c (hunk 4 above)
Comment (nbmaiti) [reply tests/unit/test_cryptoSupport.c]: done

**Pull request merged**